testsupport/signup_request.go

```diff
@@ -70,75 +70,98 @@ type SignupRequest struct {
 }
 
+// Email specifies the email address to use for the new UserSignup
 func (r *SignupRequest) Email(email string) *SignupRequest {
 	r.email = email
 	return r
 }
 
+// OriginalSub specifies the original sub value which will be used for migrating the user to a new IdP client
 func (r *SignupRequest) OriginalSub(originalSub string) *SignupRequest {
 	r.originalSub = originalSub
 	return r
 }
 
+// Resources may be called only after a call to Execute().  It returns two parameters; the first is the UserSignup
+// instance that was created, the second is the MasterUserRecord instance, HOWEVER the MUR will only be returned
+// here if EnsureMUR() was also called previously, otherwise a nil value will be returned
 func (r *SignupRequest) Resources() (*toolchainv1alpha1.UserSignup, *toolchainv1alpha1.MasterUserRecord) {
 	return r.userSignup, r.mur
 }
 
+// EnsureMUR will ensure that a MasterUserRecord is created.  It is necessary to call this function in order for
+// the Resources() function to return a non-nil value for its second return parameter.
 func (r *SignupRequest) EnsureMUR() *SignupRequest {
 	r.ensureMUR = true
 	return r
 }
 
+// WaitForMUR will wait until MasterUserRecord is created
 func (r *SignupRequest) WaitForMUR() *SignupRequest {
 	r.waitForMUR = true
 	return r
 }
 
+// GetToken may be called only after a call to Execute(). It returns the token that was generated for the request
 func (r *SignupRequest) GetToken() string {
 	return r.token
 }
 
+// ManuallyApprove if called will set the "approved" state to true after the UserSignup has been created
 func (r *SignupRequest) ManuallyApprove() *SignupRequest {
 	r.manuallyApprove = true
 	return r
 }
 
+// RequireConditions specifies the condition values that the new UserSignup is required to have in order for
+// the signup to be considered successful
 func (r *SignupRequest) RequireConditions(conditions ...toolchainv1alpha1.Condition) *SignupRequest {
 	r.conditions = conditions
 	return r
 }
 
+// VerificationRequired specifies that the "verification-required" state will be set for the new UserSignup, however
+// if ManuallyApprove() is also called then this will have no effect as user approval overrides the verification
+// required state.
 func (r *SignupRequest) VerificationRequired() *SignupRequest {
 	r.verificationRequired = true
 	return r
 }
 
+// TargetCluster may be provided in order to specify the user's target cluster
 func (r *SignupRequest) TargetCluster(targetCluster *wait.MemberAwaitility) *SignupRequest {
 	r.targetCluster = targetCluster
 	return r
 }
 
+// RequireHTTPStatus may be used to override the expected HTTP response code received from the Registration Service.
+// If not specified, here, the default expected value is StatusAccepted
 func (r *SignupRequest) RequireHTTPStatus(httpStatus int) *SignupRequest {
 	r.requiredHTTPStatus = httpStatus
 	return r
 }
 
+// Disables automatic cleanup of the UserSignup resource after the test has completed
```

Comment:
```suggestion
// DisableCleanup disables automatic cleanup of the UserSignup resource after the test has completed
```

Reply:
my bad for quickly copying/pasting without checking, but it's too bad the linter did not catch that mistake :/